posthog/special_migrations/definition.py

```diff
@@ -0,0 +1,31 @@
+from typing import Optional, Tuple
+
+from posthog.models.special_migration import SpecialMigration
+
+
+class SpecialMigrationOperation:
+    def __init__(self, sql="", database="clickhouse", timeout_seconds=60, rollback=None, resumable=False):
+        self.sql = sql
+        self.database = database
+        self.timeout_seconds = timeout_seconds
+        self.resumable = resumable
+
+
+class SpecialMigrationDefinition:
+    posthog_min_version = "0.0.0"
+    posthog_max_version = "10000.0.0"
+
+    service_version_requirements = []
+    operations = []
+
+    def is_required(self) -> bool:
+        return True
+
+    def healthcheck(self) -> Tuple[bool, Optional[str]]:
+        return (True, None)
+
+    def progress(self, migration_instance: SpecialMigration) -> int:
+        return int(100 * migration_instance.current_operation_index / len(self.operations))
+
+    def rollback(self, migration_instance: SpecialMigration) -> Tuple[bool, Optional[str]]:
```

Comment:
I've opted (for now) for a unified `rollback` function instead of each op having a rollback @tiina303 

The reason for this is that it adds complexity to the MVP. 

If an op doesn't have a rollback, do we error or do we skip? Probably error. But then, with all the rollbacks added in the ops, it gets hard to read. Plus, a group of _n_ ops can often be rolled back with _x < n_ ops. 

Ultimately I think that's what we should move towards, but this function lets people do the same thing and avoids us having to write all the handling. Happy to iterate, but will build everything else first.